viewing_party/party.py

```diff
@@ -0,0 +1,122 @@
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_to_watch):
+    for movie_data in user_data["watchlist"]:
+        if movie_data["title"] == movie_to_watch:
+            user_data["watched"].append(movie_data)
+            user_data["watchlist"].remove(movie_data)
+    
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    average_data = []
+
+    for movie_data in user_data["watched"]:
+        average_data.append(movie_data["rating"])
+
+    if not average_data:
+        return 0
+
+    return sum(average_data)/len(average_data)
+        
+def get_most_watched_genre(user_data):
+    most_watched = {}
+
+    for movie_data in user_data["watched"]:
+        if movie_data["genre"] not in most_watched:
+            most_watched[movie_data["genre"]] = 1
+        else:
+            most_watched[movie_data["genre"]] += 1
+
+    if most_watched:
+        return max(most_watched, key=most_watched.get)
+
+    return None
+    
+def get_unique_watched(user_data):
+    unique_watched = []
+    unique_watched += user_data["watched"]
+
+    for friend_data in user_data["friends"]:
+        for movie_data in friend_data["watched"]:
+            if movie_data in unique_watched:
+                unique_watched.remove(movie_data)
+
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    unique_watched = []
+
+    for friend_data in user_data["friends"]:
+        unique_watched += friend_data["watched"]
+    
+    for movie_data in user_data["watched"]:
+        if movie_data in unique_watched:
+            unique_watched.remove(movie_data)
+
+    real_unique =[]
+    [real_unique.append(x) for x in unique_watched if x not in real_unique]
+
+    return real_unique
+
+def get_available_recs(user_data):
+    available_recommendations = []
+    user_watched = []
+
+    for user_movie_data in user_data["watched"]:
+        user_watched.append(user_movie_data["title"])
+
+    for friend_data in user_data["friends"]:
+        for friend_movie_data in friend_data["watched"]:
+            if friend_movie_data["title"] not in user_watched and friend_movie_data["host"] in user_data["subscriptions"] and friend_movie_data not in available_recommendations:
+                available_recommendations.append(friend_movie_data)
+
+    return available_recommendations
+
+def get_new_rec_by_genre(user_data):
+    user_fave_genre = get_most_watched_genre(user_data)
+    user_watched = []
+
+    for user_movie_data in user_data["watched"]:
+        user_watched.append(user_movie_data["title"])
+
+    new_recs = []
+
+    for friend_data in user_data["friends"]:
+        for friend_movie_data in friend_data["watched"]:
+            if friend_movie_data["title"] not in user_watched and friend_movie_data["genre"] == user_fave_genre and friend_movie_data not in new_recs:
```

Comment:
Nice!  Small note, long lines like this can be broken up with the `\` character or by wrapping it in parens and then breaking it into multiple lines. [link](https://www.kite.com/python/answers/how-to-break-a-long-line-of-code-in-python)